catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -150,22 +186,9 @@ def open(self):
         self.log.info('Using pixel format: %s', self.current_pixel_format)
         self.cam.prop_setvalue(dcam.DCAM_IDPROP.IMAGE_PIXELTYPE, self.pixel_formats[self.current_pixel_format])
 
-        # Set device values from config file (set width and height before offsets)
-        offset_x = self.config.get('offset_x', 0)
-        offset_y = self.config.get('offset_y', 0)
-
-        self.width = self.config.get('width', self.sensor_width - offset_x)
-        self.height = self.config.get('height', self.sensor_height - offset_y)
-        self.offset_x = offset_x
-        self.offset_y = offset_y
-
-        self.gain = self.config.get('gain', 0)
-        self.exposure_time = self.config.get('exposure_time', 1000)
-        self.temperature = self.make_data_stream('temperature', 'float64', [1], 20)
-
         # Create datastreams
         # Use the full sensor size here to always allocate enough shared memory.
-        self.images = self.make_data_stream('images', 'float32', [self.sensor_height, self.sensor_width], self.NUM_FRAMES)
+        self.images = self.make_data_stream('images', 'float32', [int(self.sensor_height), int(self.sensor_width)], self.NUM_FRAMES)
```

Comment:
Hmm I am not a fan of this since it duplicates code and prevents from camera class abstraction. I think it would be better to move this to where the sensor width and height properties are retrieved, i.e. into the definition of `_create_property`. There, in the getter you can add an extra case of:
```python
if hamamatsu_property_name == (IMAGE_WIDTH or IMAGE_HEIGHT):
    return int(...)
```

Reply:
It still gives us a layer of abstraction for the data streams that we will need for the unified camera base class to be able to deal with orientations consistently, so yes please.